internal/config/image/cuda_image.go

```diff
@@ -38,27 +40,38 @@ const (
 // a map of environment variable to values that can be used to perform lookups
 // such as requirements.
 type CUDA struct {
+	logger logger.Interface
+
 	env    map[string]string
 	mounts []specs.Mount
+
+	isPrivileged bool
+
+	acceptDeviceListAsVolumeMounts bool
+	acceptEnvvarUnprivileged       bool
+	preferredVisibleDeviceEnvVars  []string
 }
 
 // NewCUDAImageFromSpec creates a CUDA image from the input OCI runtime spec.
 // The process environment is read (if present) to construc the CUDA Image.
-func NewCUDAImageFromSpec(spec *specs.Spec) (CUDA, error) {
+func NewCUDAImageFromSpec(spec *specs.Spec, opts ...Option) (CUDA, error) {
 	var env []string
 	if spec != nil && spec.Process != nil {
 		env = spec.Process.Env
 	}
 
-	return New(
+	specOpts := []Option{
 		WithEnv(env),
 		WithMounts(spec.Mounts),
-	)
+		WithPrivileged(IsPrivileged((*OCISpec)(spec))),
+	}
+
+	return New(append(opts, specOpts...)...)
```

Comment:
One thing about this behaviour is that it ASSUMES that the env and mounts from the spec are preferred. It would be good to add a test for this -- although that's not a blocker for this PR.

Reply:
it does, but then the return appends the opts, so if the caller of `NewCUDAImageFromSpec` set's the `WithPreferred...` that will be present during the construction of the image

Reply:
I'm speaking specifically of `env` and `mounts`, not `preferredVisibleDevicesEnvVars`. The point of a test is to ensure that LATER modifications don't change behaviour.

Reply:
TODO: Add unit test for `NewCUDAImageFromSpec`